backend/b2/b2.go

```diff
@@ -1828,6 +1860,9 @@ func (o *Object) Update(ctx context.Context, in io.Reader, src fs.ObjectInfo, op
 	if o.fs.opt.Versions {
 		return errNotWithVersions
 	}
+	if o.fs.opt.VersionAt != "" {
```

Comment:
I would set `opt.Versions` in `NewFs` if `VersionAt` is specified. Then we can do away with all these extra checks.

Reply:
I'm worried that might give incorrect behavior in certain obscure situations. When `opt.Versions` is set, various methods change their behavior because they expect filenames to have the version suffixes, which is not the case in version-at mode.

`getMetaDataListing` will try to strip the version suffix if `opt.Versions` is set, so if a regular file with such a version suffix happens to be present, unexpected behavior will occur. Instead it should call list as normal, which will return the correct result.

Reply:
OK fair enough!